lib/iris/cube.py

```diff
@@ -775,6 +747,9 @@ def __init__(
         if isinstance(data, str):
             raise TypeError("Invalid data type: {!r}.".format(data))
 
+        # Configure the metadata manager.
+        self._metadata = MetadataFactory(CubeMetadata)
```

Comment:
Could we call these `self._metadata_manager`?

Currently we have:
`self.metadata `- returns an instance of BaseMetadata which is a souped up namedtuple
`self._metadata `- returns the instance that MetadataFactory returns.
I find that a bit confusing.

Another good thing about `self._metadata_manager` is follows a similar naming pattern to `self._data_manager`